lib/drivers/abbott/abbottFreeStyleLibre.js

```diff
@@ -93,32 +98,59 @@ export default function (config) {
     fetchData(progress, data, cb) {
       progress(0);
 
-      const getterFunctions = [
-        (callback) => { protocol.setCompression(COMPRESSION.ENABLED, callback); },
-        (callback) => { protocol.getDbSchema(callback); },
-        (callback) => { protocol.getCfgSchema(callback); },
-        (callback) => { protocol.getDateTime(callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.METER_FACTORY_CONFIGURATION, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.METER_SETTINGS, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.USER_PATIENT_CONFIGURATION, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.USER_PATIENT_SETTINGS, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.INSULIN_SETTINGS, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.REMINDER_STRING, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.SMART_TAG_NOTES, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.STORED_SENSOR_INFORMATION, callback); },
-        (callback) => { protocol.getCfgData(CFG_TABLE_ID.REMINDER_DATA, callback); },
-        (callback) => { protocol.getDatabase(DB_TABLE_ID.GLUCOSE_RESULT, callback); },
-        (callback) => { protocol.getDatabase(DB_TABLE_ID.RAPID_ACTING_INSULIN, callback); },
-        (callback) => { protocol.getDatabase(DB_TABLE_ID.HISTORICAL_DATA, callback); },
-        (callback) => { protocol.getDatabase(DB_TABLE_ID.EVENT, callback); },
-      ];
+      let getterFunctions;
+      if (data.deviceInfo.productId === FSLIBRE_PRO_PRODUCT_ID) {
+        getterFunctions = [
+          (callback) => { protocol.getDbSchema(callback); },
+          (callback) => { protocol.getCfgSchema(callback); },
+          (callback) => { protocol.getDateTime(callback); },
+          (callback) => {
+            protocol.getCfgData(CFG_TABLE_ID.METER_FACTORY_CONFIGURATION, callback);
+          },
+          (callback) => { protocol.getCfgData(CFG_TABLE_ID.METER_SETTINGS, callback); },
+          (callback) => { protocol.getCfgData(CFG_TABLE_ID.USER_PATIENT_CONFIGURATION, callback); },
+          (callback) => { protocol.getCfgData(CFG_TABLE_ID.USER_PATIENT_SETTINGS, callback); },
+          (callback) => { protocol.getDatabase(DB_TABLE_ID.GLUCOSE_RESULT, callback); },
+          (callback) => { protocol.getDatabase(DB_TABLE_ID.HISTORICAL_DATA, callback); },
+          (callback) => { protocol.getDatabase(DB_TABLE_ID.EVENT, callback); },
+        ];
+      } else {
+        getterFunctions = [
+          (callback) => { protocol.setCompression(COMPRESSION.ENABLED, callback); },
```

Comment:
Are we sure that there is no compression possible for Libre Pro?

Reply:
that's where the upload got stuck with a timeout. 
the Pro does not answer to this command at all.
interestingly the Pro has an older firmware version (1.x) than the non-Pro (2.x)...
but the compression does not make a big difference for the pro anyhow, since there are mostly just historical glucose data sets on it, which don't include to many zeros and the compression is a very basic run-length encoding just for zeros...

Reply:
:+1: